src/vm.js

```diff
@@ -227,7 +227,8 @@ function VM(zone, name) {
      * // If the callback is omitted, we'll return a Promise.
      * //-
      * vm.getMetadata().then(function(data) {
-     *   const metadata = data[0];
+     *   const operation = data[0];
```

Comment:
Thank you, but we have a misunderstanding in our terminology here! `metadata` is a term we use throughout all of the `@google-cloud/*` repos to mean "the representation of this resource as the API sees it". In this case, the metadata isn't the internal `metadata` property which can hold user-defined key/vals, but the [Instance resource](https://cloud.google.com/compute/docs/reference/v1/instances) representation as a whole, which includes things like the labels, machine type, status, creation timestamp, and more.

Confusingly, the `setMetadata` method only allows custom user-defined key/vals, as the other properties returned in `getMetadata` cannot be changed.

I don't think we can change the behavior of the methods, but adding a line of documentation here could be helpful:

```js
const metadata = data[0];
const apiResponse = data[1];

// Custom metadata and predefined keys.
const customMetadata = metadata.metadata;
```

(Better comments/suggestions welcome!)

Reply:
Ah! That makes sense to me. The confusion really surrounded the `getMetadata` versus the `setMetadata` as you called out the difference.

Perhaps in the longer term it could be renamed -- at least in client libraries -- to `setCustomMetadata`? Then the clients could even provide a helper method that simply wraps `getMetadata` for `getCustomMetadata` to attempt to remove the confusion?

I'll make the changes to match your suggestion though. With that detail, it makes sense now.